### PR TITLE
Deploy fetcher

### PR DESCRIPTION
**Pull request by DevAseel**
**Summary of change:**
- Added a new micro-service, deployFetcher, which uses the casper sdk method getDeployInfo().

**List of tasks:**
- [x] Defined types.
- [x] Defined interfaces.
- [x] Defined class.
- [x] Created unit test.
- [x] Generated the docs. 
- [x] Updated the documentation.

**Steps for testing**
- Make sure your local git is up to date
```bash
git pull
```

- Checkout the branch `feature/deploy-fetcher`
```bash
git checkout feature/deploy-fetcher
```

- Run the test across the test-net
```bash
npm run test
```

- Confirm deployFetcher test has passed.
- Run the docs using live server and make sure it have been updated.

### CODE REVIEW

File: test/mocks/MockCasperBlockchain.ts [hunk 1]
```diff
@@ -58,34 +60,55 @@ export class MockCasperBlockchain implements IBlockchain {
...
     }
 
     return undefined;
   }
+
+  async getDeployByHash(hash: string): Promise<GetDeployResult> {
+    if (this.getSuccess('getDeployByHash')) {
+      if (!!this.deploys) {
+        for (const deploy of this.deploys) {
+          if (!!deploy && !!deploy.fields && !!deploy.fields.header && !!deploy.fields.header.hash) {
```
Comment (specialistvlad) [inline test/mocks/MockCasperBlockchain.ts:96]: Please use here `?.' operator
https://bobbyhadz.com/blog/typescript-question-mark-dot

Comment (specialistvlad) [reply test/mocks/MockCasperBlockchain.ts:96]: Check files for same improvement

Comment (DevAseel) [reply test/mocks/MockCasperBlockchain.ts:96]: @jjfluid please advise on the above.

**Review by specialistvlad — APPROVED** [review]

**Review by jjfluid — APPROVED** [review]
This one was good to go.

File: test/mocks/MockCasperBlockchain.ts (hunk 1 above)
Comment (jjfluid) [reply test/mocks/MockCasperBlockchain.ts:96]: its fine. how many users will be using node12? if you think slim to none, we can drop it from the test runner

Comment (jjfluid) [reply test/mocks/MockCasperBlockchain.ts:96]: our backend is 14+

Comment (DevAseel) [reply test/mocks/MockCasperBlockchain.ts:96]: I think it's safe, but we have to specify the node version in requirements file.

**Pull request merged**